Sources/IRGen/Function/IULIAExpression.swift

```diff
@@ -104,18 +104,42 @@ struct IULIAPropertyAccess {
   var lhs: Expression
   var rhs: Expression
   var asLValue: Bool
-  
+
   func rendered(functionContext: FunctionContext) -> String {
     let environment = functionContext.environment
     let scopeContext = functionContext.scopeContext
     let enclosingTypeName = functionContext.enclosingTypeName
     let isInStructFunction = functionContext.isInStructFunction
 
     var isMemoryAccess: Bool = false
-    
+
     let lhsType = environment.type(of: lhs, enclosingType: enclosingTypeName, scopeContext: scopeContext)
-    let rhsOffset = IULIAPropertyOffset(expression: rhs, enclosingType: lhsType).rendered(functionContext: functionContext)
-    
+
+    let rhsOffset: String
+    // Special cases.
+    switch lhsType {
+    case .fixedSizeArrayType(_, let size):
+      if case .identifier(let identifier) = rhs, identifier.name == "size" {
+        return "\(size)"
+      } else {
+        fatalError()
+      }
+    case .arrayType(_):
+      if case .identifier(let identifier) = rhs, identifier.name == "size" {
+        rhsOffset = "0"
+      } else {
+        fatalError()
+      }
+    case .dictionaryType(_):
+      if case .identifier(let identifier) = rhs, identifier.name == "size" {
+        rhsOffset = "0" // TODO: check!
```

Comment:
Is this a leftover fragment or do you still need to add this check?

Reply:
This is a leftover, I will remove it.